pkg/git/git.go

```diff
@@ -244,3 +253,26 @@ func userHasKnownHostsFile(logger *zap.SugaredLogger) (bool, error) {
 	f.Close()
 	return true, nil
 }
+
+func validateGitAuth(logger *zap.SugaredLogger, url string) error {
+	homeenv := os.Getenv("HOME")
+	sshCred := true
+	if _, err := os.Stat(homeenv + "/.ssh"); os.IsNotExist(err) {
+		sshCred = false
+	}
+	urlSSHFormat := validateGitSSHURLFormat(url)
+	if sshCred && !urlSSHFormat {
+		logger.Warnf("The given URL does not match the ssh credential")
```

Comment:
It would be great to include the URL that did not match expected format.  `The given URL %q does not match...`